lib/playground.dart

```diff
@@ -397,8 +402,23 @@ class Playground implements GistContainer {
     }
   }
 
+  void _checkForHtml() {
+    if (_context.htmlDocument.value.trim().isEmpty && !htmlIsEmpty) {
+      htmlIsEmpty = true;
+      querySelector("#consoletab").style.display = "none";
+      querySelector("#resulttab").style.display = "none";
+      _outputpanel.style.display = "block";
+      _frame.style.display = "none";
+    } else if (_context.htmlDocument.value.trim().isNotEmpty && htmlIsEmpty){
+      htmlIsEmpty = false;
+      querySelector("#consoletab").style.display = "block";
+      querySelector("#resulttab").style.display = "block";
+      _toggleResultTab();
+    }
```

Comment:
Could you put 1 line comments as to what the state machine here is?

I think the first one is "currently showing HTML -> Not showing HTML "
and the second is 
"not showing HTML -> showing HTML"

If this is the case, the method should probably be renamed as it has side effects. Consider e.g.:

_updateResultTabVisible()